Quantum_Circuit_Design/helloworld_DQN_single_file.py

```diff
@@ -716,11 +696,16 @@ def valid_agent(env_class, env_args: dict, net_dims: List[int], agent_class, act
 def train_dqn_for_quantum(gpu_id: int = 0):
     # Replace env_class with QuantumEnv
     env_class = QuantumEnv
+    target_circuit = text_circuit1
+    num_qubits = 3
+    
     env_args = {
         'env_name': 'Quantum-v1',  # Custom environment name
-        'state_dim': 16,  # QuantumEnv outputs a flattened unitary of size 2 * 4 (assuming 2 qubits)
-        'action_dim':14,  # Number of possible quantum actions (gates)
+        'state_dim': 4**num_qubits,  # QuantumEnv outputs a flattened unitary of size 2 * 4 (assuming 2 qubits)
+        'action_dim':2*num_qubits+num_qubits*(num_qubits-1),  # Number of possible quantum actions (gates)
```

Comment:
Currently action dimension need to be changed manually for each universal gate set. Requires more change for automatically calculated action dimension